src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,263 @@
+        :label="`${props.badgeText}` ? iconTitle : tooltipText"
+        max-width="500px"
+        placement="bottomStart"
+        position-fixed
+      >
+        <KClipboardProvider v-slot="{ copyToClipboard }">
+          <CopyIcon
+            class="text-icon"
+            data-testid="copy-to-clipboard"
+            :hide-title="!!copyTooltip || undefined"
+            role="button"
+            :size="KUI_ICON_SIZE_40"
+            tabindex="0"
+            @click.stop="copyIdToClipboard(copyToClipboard)"
+          />
+        </KClipboardProvider>
+      </KTooltip>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from 'vue'
+import { computed, ref } from 'vue'
+import { CopyIcon } from '@kong/icons'
+import KClipboardProvider from '@/components/KClipboardProvider'
+import KTooltip from '@/components/KTooltip/KTooltip.vue'
+import { KUI_ICON_SIZE_40 } from '@kong/design-tokens'
+
+const props = defineProps({
+  /**
+    * Text displayed before the copyable text when
+    * `badge` is true
+    */
+  badgeText: {
+    type: String,
+    default: '',
+  },
+  /**
+    * The copyable text
+    */
+  text: {
+    type: String,
+    required: true,
+  },
+  /**
+    * Tooltip text displayed on hover over the `text`
+    */
+  textTooltip: {
+    type: String,
+    default: '',
+  },
+  /**
+    * Tooltip text displayed on hover over copy button
+    */
+  copyTooltip: {
+    type: String,
+    default: 'Copy',
+  },
+  /**
+    * Formatting for copyable text (default, hidden, redacted, deleted)
+    */
+  format: {
+    type: String as PropType<'default' | 'hidden' | 'redacted' | 'deleted'>,
+    required: false,
+    default: 'default',
+    validator: (val: string) => ['default', 'hidden', 'redacted', 'deleted'].includes(val),
+  },
+  /**
+    * Whether or not to display as a badge
+    */
+  badge: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+    * False if `badge` is true
+    */
+  monospace: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+    * Whether or not the text should be truncated
+    */
+  truncate: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+    * Tooltip text displayed on successful copy
+    */
+  successTooltip: {
+    type: String,
+    default: 'Copied',
+  },
+  /**
+    * Number of characters to truncate at
+    */
+  truncationLimit: {
+    type: Number,
+    default: 8,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'copy', val: string): void
+}>()
+
+const hasSuccessTooltip = computed((): boolean => !!(props.copyTooltip && props.successTooltip))
+const tooltipText = ref(props.copyTooltip)
+
+const badgeTextLabel = computed((): string => {
+  if (!props.badgeText) {
+    return ''
+  }
+
+  // strip trailing colon from label if one exists
+  return props.badgeText.replace(/:$/, '')
+})
+
+const iconTitle = computed(() => {
+  if (props.badgeText) {
+    return `Copy ${badgeTextLabel.value}`
+  }
+  return 'Copy'
```

Comment:
question: this hard-coded text will be an issue when translation is a thing. Guessing that doesn't matter for now?

Reply:
KSelect/KPrompt returns hard-coded text too, and there is no usage of i18n in kongponents, so we should be good.